CogniteSdk/test/csharp/DataModels.cs

```diff
@@ -641,5 +642,54 @@ public async Task TestRunQuery()
             }
 
         }
+
+        [Fact]
+        public async Task TestCustomResource()
+        {
+            var resource = new TestResource(tester);
+            await resource.UpsertAsync(new[] {
+                new SourcedNodeWrite<TestItem> {
+                    ExternalId = "node9",
+                    Space = tester.TestSpace,
+                    Properties = new TestItem {
+                        Prop = "test",
+                        IntProp = 123
+                    }
+                }
+            }, new UpsertOptions());
+
+            var retrieved = await resource.RetrieveAsync(new[] { new InstanceIdentifier {
+                InstanceType = InstanceType.node,
+                Space = tester.TestSpace,
+                ExternalId = "node9"
+            }});
+            Assert.Single(retrieved);
+            var node = retrieved.First();
+            Assert.Equal("test", node.Properties.Prop);
+            Assert.Equal(123, node.Properties.IntProp);
+
+            await resource.DeleteAsync(new[] { new InstanceIdentifier {
+                InstanceType = InstanceType.node,
+                Space = tester.TestSpace,
+                ExternalId = "node9"
+            }});
```

Comment:
Would be good to do another retrieve after the delete, to ensure that it worked.

Reply:
Don't think deletes are immediately consistent, unfortunately.